x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/results/InferenceResults.java

```diff
@@ -8,8 +8,11 @@
 import org.elasticsearch.common.io.stream.NamedWriteable;
 import org.elasticsearch.ingest.IngestDocument;
 
+import java.util.Map;
+
 public interface InferenceResults extends NamedWriteable {
 
     void writeResult(IngestDocument document, String parentResultField);
 
+    Map<String, Object> writeResultToMap(String parentResultField);
```

Comment:
I wonder if we should implement `ToXContentObject` instead of having a bespoke method that creates a map.

Reply:
The map is converted to `DocumentField`s we couldn't do that with `ToXContentObject`

Reply:
Cool :D